homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -22,38 +25,68 @@ class MinecraftServerConfigFlow(ConfigFlow, domain=DOMAIN):
 
     async def async_step_user(self, user_input=None):
         """Handle the initial step."""
+        errors = {}
+
         if user_input is not None:
             # User inputs.
             host = user_input[CONF_HOST]
             port = user_input[CONF_PORT]
 
-            # Abort in case the host was already configured before.
-            unique_id = f"{host}-{port}"
-            await self.async_set_unique_id(unique_id)
-            self._abort_if_unique_id_configured()
+            unique_id = ""
 
-            errors = {}
+            # Check if 'host' is a valid IP address and if so, get the MAC address.
+            ip_address = None
+            mac_address = None
+            try:
+                ip_address = ipaddress.ip_address(host)
+            except ValueError:
+                # Host is not a valid IP address.
+                pass
+            else:
+                # Host is a valid IP address.
+                if ip_address.version == 4:
+                    # Address type is IPv4.
+                    mac_address = getmac.get_mac_address(ip=host)
```

Comment:
Getting the mac address is not async safe. Use `hass.async_add_executor_job`.

Reply:
```py
if ip_address.version == 4:
    params = {"ip": host}
else:
    params = {"ip6": host}
mac_address = await hass.async_add_executor_job(
    partial(getmac.get_mac_address, **params)
)
```

Reply:
Whoops 😳 Thanks for the code snippet. Didn't know about partial. Everyday I learn s.th. new here 👍